File: app/src/main/java/com/usher/demo/material/FitsSystemWindowActivity.java
package com.usher.demo.material;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.usher.demo.R;

public class FitsSystemWindowActivity extends AppCompatActivity {
    private final String EXTRA_NAME = "EXTRA_TAG";
    private final String EXTRA_VALUE_NONE = "NONE";
    private final String EXTRA_VALUE_ROOT = "ROOT";
    private final String EXTRA_VALUE_TOP = "TOP";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_fitssystemwindow);

        initView();

//        getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
//        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN | View.SYSTEM_UI_FLAG_LAYOUT_STABLE);
    }

    private void initView() {
        TextView statusTextView = findViewById(R.id.status_textview);
        statusTextView.setText(getString(R.string.fitssystemwindow_status_text, "NONE"));
        String value = getIntent().getStringExtra(EXTRA_NAME);

        if (value != null) {
            switch (getIntent().getStringExtra(EXTRA_NAME)) {
                case EXTRA_VALUE_ROOT:
                    findViewById(R.id.root_layout).setFitsSystemWindows(true);
                    statusTextView.setText(getString(R.string.fitssystemwindow_status_text, "Root View"));
                    break;
                case EXTRA_VALUE_TOP:
//                    findViewById(R.id.attr_des_textview).setFitsSystemWindows(true);
                    findViewById(R.id.top_textview).setFitsSystemWindows(true);
                    statusTextView.setText(getString(R.string.fitssystemwindow_status_text, "Top View"));

                    break;
                default:
                    break;
            }
        }

        findViewById(R.id.none_view_textview).setOnClickListener(v -> {
            Intent intent = new Intent(this, FitsSystemWindowActivity.class);
            intent.putExtra(EXTRA_NAME, EXTRA_VALUE_NONE);
            startActivity(intent);
            finish();
        });

        findViewById(R.id.root_view_textview).setOnClickListener(v -> {
            Intent intent = new Intent(this, FitsSystemWindowActivity.class);
            intent.putExtra(EXTRA_NAME, EXTRA_VALUE_ROOT);
            startActivity(intent);
            finish();
        });

        findViewById(R.id.top_view_textview).setOnClickListener(v -> {
            Intent intent = new Intent(this, FitsSystemWindowActivity.class);
            intent.putExtra(EXTRA_NAME, EXTRA_VALUE_TOP);
            startActivity(intent);
            finish();
        });
    }
}

File: app/src/main/java/com/usher/demo/material/FitsSystemWindowActivity.kt
package com.usher.demo.material

import android.content.Intent
import com.jakewharton.rxbinding4.view.clicks
import com.twigcodes.ui.util.RxUtil
import com.usher.demo.R
import com.usher.demo.base.BaseActivity
import kotlinx.android.synthetic.main.activity_fitssystemwindow.*

class FitsSystemWindowActivity : BaseActivity(R.layout.activity_fitssystemwindow) {
    companion object {
        private const val EXTRA_NAME = "EXTRA_TAG"
        private const val EXTRA_VALUE_NONE = "NONE"
        private const val EXTRA_VALUE_ROOT = "ROOT"
        private const val EXTRA_VALUE_TOP = "TOP"
    }

    override fun initView() {
        val name = when (intent.getStringExtra(EXTRA_NAME)) {
            EXTRA_VALUE_ROOT -> {
                root_layout.fitsSystemWindows = true
                getString(R.string.fitssystemwindow_root_view)
            }
            EXTRA_VALUE_TOP -> {
                top_textview.fitsSystemWindows = true
                getString(R.string.fitssystemwindow_top_view)
            }
            else -> getString(R.string.fitssystemwindow_none_view)
        }

        status_textview.text = getString(R.string.fitssystemwindow_status_text, name)

        none_view_textview.clicks()
                .take(1)
                .to(RxUtil.autoDispose(this))
                .subscribe {
                    startActivity(Intent(this, FitsSystemWindowActivity::class.java).apply {
                        putExtra(EXTRA_NAME, EXTRA_VALUE_NONE)
                    })
                    finish()
                }

        root_view_textview.clicks()
                .take(1)
                .to(RxUtil.autoDispose(this))
                .subscribe {
                    startActivity(Intent(this, FitsSystemWindowActivity::class.java).apply {
                        putExtra(EXTRA_NAME, EXTRA_VALUE_ROOT)
                    })
                    finish()
                }

        top_view_textview.clicks()
                .take(1)
                .to(RxUtil.autoDispose(this))
                .subscribe {
                    startActivity(Intent(this, FitsSystemWindowActivity::class.java).apply {
                        putExtra(EXTRA_NAME, EXTRA_VALUE_TOP)
                    })
                    finish()
                }
    }
}